src/pynwb/base.py

```diff
@@ -398,6 +398,57 @@ def get_timestamps(self):
         else:
             return np.arange(len(self.data)) / self.rate + self.starting_time
 
+    def get_starting_time(self):
+        """
+        Get the starting time of this TimeSeries in seconds.
+
+        Returns
+        -------
+        float or None
+            The starting time in seconds, or None if the TimeSeries has no data.
+        """
+        if self.num_samples is None or self.num_samples == 0:
+            return None
+        if self.fields.get('timestamps'):
+            return float(self.timestamps[0])
+        else:
+            return self.starting_time
+
+    def get_duration(self):
+        """
+        Get the duration of this TimeSeries in seconds.
+
+        Returns the time span from the first sample to the last sample.
+        For a single sample, returns 0.
+
+        Returns
+        -------
+        float or None
+            The duration in seconds, or None if the TimeSeries has no data.
+
+        Notes
+        -----
+        For rate-based TimeSeries: duration = (n - 1) / rate
+        For timestamp-based TimeSeries: duration = timestamps[-1] - timestamps[0]
+
+        The duration represents the time span between sample times, not the total
+        recording time. If you need to account for the last sample's duration
+        (e.g., for continuous recordings), add 1/rate manually.
+        """
+        n = self.num_samples
+        if n is None or n == 0:
+            return None
```

Comment:
```suggestion
        if n is None or n == 0:
            return 0
```
I think a duration should be 0 if the TimeSeries is empty